File: 0728-self-dividing-numbers/0728-self-dividing-numbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        result=[]
        count=0
        for num in range (left,right+1):
            num1=str(num)
            if '0' in num1:
                continue
            else:
                for i in range (0,len(num1)):
                    if num%int(num1[i])!=0:
                        count+=1
            if count==0:
                result.append(num)
            count=0
        return result
                        

            

File: 0728-self-dividing-numbers/0728-self-dividing-numbers.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* selfDividingNumbers(int left, int right, int* returnSize) {
    int *arr = (int*)malloc((right - left + 1)*(sizeof(int)));
    int ind=0;
    for (int i=left; i<right+1; i++){
        bool check=true;
        int n=i;
        while (n>0){
            
            int digit = n%10;
            if (digit==0 || i%digit !=0) {
                check=false; 
                break;
            }
            n/=10;
                
            
        }
        if (check) {
            arr[ind++]=i;
        }

    }
    *returnSize=ind;
    return arr;
}